sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -312,3 +315,90 @@ case class Logarithm(left: Expression, right: Expression)
     """
   }
 }
+
+case class Round(child: Expression, scale: Expression) extends Expression {
+
+  def this(child: Expression) = {
+    this(child, Literal(0))
+  }
+
+  def children: Seq[Expression] = Seq(child, scale)
+
+  def nullable: Boolean = true
+
+  private lazy val scaleV = scale.asInstanceOf[Literal].value
+  private lazy val _scale = if (scaleV != null) scaleV.asInstanceOf[Int] else 0
+
+  override lazy val dataType: DataType = {
+    child.dataType match {
+      case StringType | BinaryType => DoubleType
+      case DecimalType.Fixed(p, s) => DecimalType(p, _scale)
+      case t => t
+    }
+  }
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    child.dataType match {
+      case _: NumericType | NullType | BinaryType | StringType => // satisfy requirement
+      case dt =>
+        return TypeCheckFailure(s"Only numeric, string or binary data types" +
+          s" are allowed for ROUND function, got $dt")
+    }
+    scale match {
+      case Literal(value, LongType) =>
+        if (value.asInstanceOf[Long] < Int.MinValue || value.asInstanceOf[Long] > Int.MaxValue) {
+          return TypeCheckFailure("ROUND scale argument out of allowed range")
+        }
+      case Literal(_, _: IntegralType) | Literal(_, NullType) => // satisfy requirement
+      case child =>
+        if (child.find { case _: AttributeReference => true; case _ => false } != None) {
```

Comment:
@rxin This is actually a general case for requiring the child to be `Literal`, probably we'd better not do check that in `checkInputDataTypes()`, but in type coercion of the expression framework. Or, we can take it as the normal expression, it should not reduce the performance of execution, as the CPU branch prediction technology.